pandas/core/generic.py

```diff
@@ -5907,6 +5908,137 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_type(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
+        result = self
+        target_int_dtype = "Int64"
+        if is_platform_32bit():
+            target_int_dtype = "Int32"
+
+        if is_object_dtype(dtype):
+            new_dtype = lib.infer_dtype(self)
+            if (
+                new_dtype != "string"
+                and new_dtype != "boolean"
+                and new_dtype != "integer"
+            ):
+                new_dtype = dtype
+            else:
+                changeit = True
+        elif is_integer_dtype(dtype):
+            if not is_extension_array_dtype(dtype):
+                new_dtype = "integer"
+                changeit = True
+        elif is_numeric_dtype(dtype):
+            new_dtype = lib.infer_dtype(list(self))
```

Comment:
Does this ever return something else as "floating"? (or "empty")

Reply:
@jorisvandenbossche  Not sure I understand your question.  By "this", you mean `lib.infer_dtype`? The infer could return "integer" or "mixed_integer", etc.

Reply:
Yes, but integers (and integers as object) already were covered in previous `if/elif` branches, so the remaining things passing `is_numeric_dtype` will be floating columns, I would think? And then, won't `infer_dtype` always return "floating"?

Reply:
@jorisvandenbossche  Yes, you're right.  At some point, I thought I had a case where that didn't happen, but that is no longer true, so I've changed it for the next commit.